flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleMaster.java

```diff
@@ -121,4 +122,21 @@ default CompletableFuture<Collection<PartitionWithMetrics>> getAllPartitionWithM
             JobID jobId) {
         return CompletableFuture.completedFuture(Collections.emptyList());
     }
+
+    /**
+     * Whether the shuffle master supports taking snapshot in batch scenarios, which will be used
+     * when enable Job Recovery. If it returns true, we will call {@link #snapshotState} to take
```

Comment:
which will be used when enable Job Recovery
-> if `execution.batch.job-recovery.enabled` is true.

we
-> Flink